UMN-AutoPackager/public/Get-NewPackageFile.ps1

```diff
@@ -0,0 +1,141 @@
+function Get-NewPackageFile {
+<#
+.SYNOPSIS
+Cmdlet that downloads a file from a web server or file share and saves it to the specified location.
+.DESCRIPTION
+Get-NewPackageFile is a function that when provided with a Source file path, UNC, or URI (as a string)
+and a Destination path or UNC (as a string) will copy or download the Source file to the Destination.
+Characters of a space (" ") or an 'escaped' space ("%20") are replaced with an underscore ("_") when
+saved to the Destination.
+.PARAMETER Source
+A string that must be a file path, UNC, or URI of the file to be saved to the Destination location.
+.PARAMETER Destination
+A string that must be a file path or UNC where the Source file will be saved.
+.EXAMPLE
+Get-NewPackageFile -Source "C:\Test\File1.txt" -Destination "D:\Temp\File2.txt"
+.EXAMPLE
+Get-NewPackageFile -Source "C:\Test\File1.txt" -Destination "D:\Temp\"
+.EXAMPLE
+Get-NewPackageFile -Source "C:\Test\File1.txt" -Destination "\\fileserver\Temp\File1.txt"
+.EXAMPLE
+Get-NewPackageFile -Source "\\fileserver\Test\File1.txt" -Destination "D:\Temp\File1.txt"
+.EXAMPLE
+Get-NewPackageFile -Source "\\fileserver1\Test\File1.txt" -Destination "\\fileserver2\Temp\File1.txt"
+.EXAMPLE
+Get-NewPackageFile -Source "https://webserver.xyz/Test/File1.txt" -Destination "C:\Temp\File1.txt"
+.EXAMPLE
+Get-NewPackageFile -Source "https://webserver.xyz/Test/File1.txt" -Destination "\\fileserver\Temp\File1.txt"
+.INPUTS
+String
+.NOTES
+If the Destination parameter is only a file path and does not include a filename, the filename obtained from
+the Source parameter is used during the copy. Requires use of the "Get-RedirectedUri" function.
+#>
+    
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory = $true,
+            HelpMessage = "Enter a valid Source file path, UNC, or URI.")]
+        [ValidateNotNullOrEmpty()]
+        [string]$Source,
+
+        [Parameter(Mandatory = $true,
+            HelpMessage = "Enter a valid Destination file path or UNC.")]
+        [ValidateNotNullOrEmpty()]
+        [string]$Destination
+    )
+
+    process {
+        try {
+            Write-Verbose "Verbose output enabled"
```

Comment:
You should use the -Message "blah" on Write-Verbose calls to make things clear.  I'm only going to add this comment here, but applies to all instances of Write-Verbose.

I would also clean up the number of Write-Verbose instances you have.  There are way too many.  These execute every time the function is run but are outputted to null if the -Verbose switch isn't enabled.  This still takes up processing time.

Reply:
Good info! Reducing and consolidating the number of Write-Verbose instances, and adding "-Message" to those that remain. Resolving coversation.